ui/app/AppLayouts/Wallet/popups/buy/BuyCryptoModal.qml

```diff
@@ -0,0 +1,140 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.0
+import QtQml.Models 2.14
+import SortFilterProxyModel 0.2
+
+import StatusQ.Popups 0.1
+import StatusQ.Popups.Dialog 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+import StatusQ 0.1
+
+import utils 1.0
+
+import AppLayouts.Wallet.controls 1.0
+import AppLayouts.Wallet.adaptors 1.0
+import AppLayouts.Wallet.panels 1.0
+
+StatusStackModal {
+    id: root
+
+    required property BuyCryptoParamsForm buyCryptoInputParamsForm
+    required property BuyCryptoModalAdaptor buyCryptoAdaptor
+    
+    QtObject {
+        id: d
+        readonly property var buyOnMercuryoButton: StatusButton {
+            height: finishButton.height
+            visible: !!root.replaceItem
+            borderColor: "transparent"
+            text: qsTr("Buy via %1").arg(!!root.buyCryptoAdaptor.selectedProvider ? root.buyCryptoAdaptor.selectedProvider.name: "")
+            loading: root.buyCryptoAdaptor.urlIsBeingFetched
+            onClicked: {
+                if(!!root.buyCryptoAdaptor.selectedProvider && !!root.buyCryptoAdaptor.selectedToken) {
+                    root.buyCryptoAdaptor.fetchProviderUrl(
+                                root.buyCryptoInputParamsForm.selectedProviderId,
+                                buyCryptoProvidersListPanel.currentTabIndex,
+                                root.buyCryptoInputParamsForm.selectedWalletAddress,
+                                root.buyCryptoInputParamsForm.selectedNetworkChainId,
+                                root.buyCryptoAdaptor.selectedToken.symbol
+                                )
+                }
+            }
+            enabled: root.buyCryptoInputParamsForm.filledCorrectly
+        }
+    }
+
+    width: 560
+    height: 515
+    padding: Style.current.xlPadding
+    stackTitle: qsTr("Buy assets for %1").arg(!!buyCryptoAdaptor.selectedAccount ? buyCryptoAdaptor.selectedAccount.name: "")
+    rightButtons: [d.buyOnMercuryoButton, finishButton]
+    finishButton: StatusButton {
+        text: qsTr("Done")
+        onClicked: root.close()
+    }
+
+    Connections {
+        target: root.buyCryptoAdaptor
+        function onProviderUrlReady(url) {
```

Comment:
the URL will be empty if there's some error when fetching it. Ideally we show some error message, but for now at least close the popup instead of opening with an empty URL